packages/venia-concept/src/components/Gallery/item.js

```diff
@@ -18,29 +18,36 @@ const ItemPlaceholder = ({ children, classes }) => (
 
 class GalleryItem extends Component {
     static propTypes = {
-        classes: PropTypes.shape({
-            image: PropTypes.string,
-            image_pending: PropTypes.string,
-            imagePlaceholder: PropTypes.string,
-            imagePlaceholder_pending: PropTypes.string,
-            images: PropTypes.string,
-            images_pending: PropTypes.string,
-            name: PropTypes.string,
-            name_pending: PropTypes.string,
-            price: PropTypes.string,
-            price_pending: PropTypes.string,
-            root: PropTypes.string,
-            root_pending: PropTypes.string
+        classes: shape({
+            image: string,
+            image_pending: string,
+            imagePlaceholder: string,
+            imagePlaceholder_pending: string,
+            images: string,
+            images_pending: string,
+            name: string,
+            name_pending: string,
+            price: string,
+            price_pending: string,
+            root: string,
+            root_pending: string
         }),
-        item: PropTypes.shape({
-            key: PropTypes.string.isRequired,
-            image: PropTypes.string,
-            name: PropTypes.string,
-            price: PropTypes.string
+        item: shape({
+            id: number.isRequired,
+            name: string.isRequired,
+            small_image: string.isRequired,
+            price: shape({
+                regularPrice: shape({
+                    amount: shape({
+                        value: number.isRequired,
+                        currency: string.isRequired
+                    }).isRequired
+                }).isRequired
+            }).isRequired
```

Comment:
If an item doesn't have a price, will this full shape be present anyway? Will it just be `$0.00` all the way down?

Reply:
May just be something we have to find out. I'd expect it to always be present, but the schema doesn't have enough info to really tell me.

It's not marked as a non-nullable field in the schema, but I haven't seen that team make much use of non-nullable fields anywhere.

Pinging @eric-bohanon

Reply:
We are currently not using non-nullable fields in GraphQL schema, but it makes sense to consider this as an improvement. @DrewML feel free to create an issue in magento/graphql-ce project listing any fields that you believe should be made non-nullable.